integration-libs/cds/src/profiletag/services/profile-tag-push-events.service.ts

```diff
@@ -350,6 +355,25 @@ export class ProfileTagPushEventsService {
     );
   }
 
+  protected cartChangedEvent(): Observable<ProfileTagPushEvent> {
```

Comment:
After you message again in the thread, I took a peek at the code instead of someone already approving, but wouldn't the snippet achieve the same result? 

Instead of treating the pipe like a single function, let's treat it as a 'chain of async tasks'. Basically, streams in a declarative manner. Let me know if it works for you~ (also if you try to find in the codebase, we don't usually go for pipe(() => ...)
```
  protected cartChangedEvent(): Observable<ProfileTagPushEvent> {
    return merge(
      this.eventService.get(CartAddEntrySuccessEvent),
      this.eventService.get(CartUpdateEntrySuccessEvent),
      this.eventService.get(CartRemoveEntrySuccessEvent)
    ).pipe(
      withLatestFrom(
        this.activeCartFacade.getActive().pipe(
          distinctUntilChanged(),
          filter((cart) => cart.code !== undefined)
        )
      ),
      map(([_, cart]) => new CartSnapshotPushEvent({ cart }))
    );
  }
```

P.S 
1. you can always filter for cart.code to be undefined, but wouldn't we say that if cart returns an empty object, then it means we cu
<img width="852" alt="Screen Shot 2022-04-29 at 11 24 03 AM" src="https://user-images.githubusercontent.com/29099466/165975404-be942988-8560-4e56-9a95-bc5edfc2b461.png">
rrently have no active cart, which in turns mean, there is no code as having a cart would mean that there is an attached code to it always I believe
2. don't think you need distinctUntilChanged when the service already does it 

screenshot from `feature-libs/cart/base/core/facade/active-cart.service.ts`

Reply:
Hi Brian! Thanks for having a look :) 

However, I am afraid that the solution that you are proposing won't work because of the following reasons:

1. When the user adds an item to the cart for the first time, the current code sends 2 CartSnapshot events one with productSkus: [] and the other with the new item in the cart. The filtering for the cart.code was done to avoid sending events when the cart is not initialized yet and that was the case. However, with the change u proposed, only one event is sent with empty skus thus not capturing the cart state on AddToCart event.
2. When the user removes the last item from the cart, as of now a CartSnapshot event is sent with empty skus. However, with the change that u proposed, its sending it with the sku id 🤔 

I do agree with point 2 :) Let me know what u think of this :)

Reply:
double checked and worked on it~